### PR TITLE
Add transform for DSpace METS records

### PR DESCRIPTION
**Pull request by hakbailey**
#### What does this PR do?
Adds a DSpace METS transform class and CLI source option.

#### Helpful background context
This transform will enable us to ingest DSpace@MIT records into TIMDEX via the new ETL workflow.

#### How can a reviewer manually see the effects of these changes?
Run the transform locally using the `dspace-sample-harvest-mets-2022-06-13.xml` file in the dev S3 bucket, e.g.:
```
pipenv run transform -i <S3 URL for file above> -o output/dspace-mets-samples-timdex.json -s dspace
```

#### What are the relevant tickets?
* https://mitlibraries.atlassian.net/browse/RDI-120

#### Includes new or updated dependencies?
NO

#### Developer
- [x] All new ENV is documented in README
- [x] Stakeholder approval has been confirmed (or is not needed)

#### Code Reviewer
- [ ] The commit message is clear and follows our guidelines
      (not just this pull request message)
- [ ] There are appropriate tests covering any new functionality
- [ ] The documentation has been updated or is unnecessary
- [ ] The changes have been verified
- [ ] New dependencies are appropriate or there were no changes

### CODE REVIEW

File: transmogrifier/sources/dspace_mets.py [hunk 1]
```diff
@@ -0,0 +1,268 @@
...
+            if file and file.get("MIMETYPE"):
+                kwargs.setdefault("file_formats", []).append(file["MIMETYPE"])
+
+        # format
+        kwargs["format"] = "electronic resource"
+
+        # funding_information: relevant field in DSpace (dc.description.sponsorship) is
+        # not mapped to the OAI-PMH METS output.
+
+        # holdings field not used in DSpace
```
Comment (ehanson8) [inline transmogrifier/sources/dspace_mets.py:158]: There’s a lot of comments here and I understand some when clarification is needed (why a field may exist in DC but not the METS export) but some like this feel like what Clean Code was trying to dissuade us from creating and won’t likely be useful to future developers. If we want to retain this info, I think there are better methods of documentation than interspersed with the code. Non-blocking but I’d recommend a second look at whether all these are necessary

**Review by ehanson8 — APPROVED** [review]
Code works as expected and I'll use the local testing method with the s3 files going forward with my PRs on this repo. Some non-blocking suggestions/comments for consideration before merging

File: transmogrifier/helpers.py [hunk 2]
```diff
@@ -8,6 +8,56 @@
 from transmogrifier.models import TimdexRecord
 
 
+def generate_citation(extracted_data: dict) -> str:
```
[the remark this answers is not in the document]
Comment (hakbailey) [reply transmogrifier/helpers.py:11]: Already did :) see RDI-141. We can update that to include dspace dim if you don't want to include it in your current pr.

File: transmogrifier/sources/dspace_mets.py [hunk 3]
```diff
@@ -0,0 +1,268 @@
+"""DSpace METS XML transform module."""
+import logging
+from typing import Iterator
+
+from bs4 import Tag
+
+import transmogrifier.models as timdex
```
[the remark this answers is not in the document]
Comment (hakbailey) [reply transmogrifier/sources/dspace_mets.py:7]: Interesting. I think the code is much more clear this way (what's the difference between `identifier` and `Identifier`?). I think I would agree with you if we named all the variables something like `source_identifier` and `timdex_identifier` but this means we don't have to. I also dislike explicitly importing every single class used, especially when there are so many. And, I don't think this is likely to be an issue in this codebase, but theoretically a class or function name could be reused in multiple modules so namespacing like this is helpful for disambiguation.

File: transmogrifier/sources/dspace_mets.py (hunk 1 above)
Comment (hakbailey) [reply transmogrifier/sources/dspace_mets.py:158]: Hmm. I think this is exactly what Clean Code was promoting: comments aren't for explaining _what_ the code does, but rather _why_. We don't expect those initial mapping spreadsheets to stick around, so documenting why some fields are mapped and some aren't is useful to have here. Knowing why `holdings` isn't mapped is just as useful as knowing why `funding_information` isn't mapped, regardless of the reason. And I'm not sure where else would be a better place for it, as going forward this codebase is the official source of truth for all of these data mappings.

**Comment by hakbailey** [comment]
Force push to squash commits.

**Pull request merged**